### ISSUE TITLE
NavigationView requires context to be an instance of FragmentActivity

### ISSUE THREAD

**Issue by Subtle-fox** [mapbox/mapbox-navigation-android#2777]
Code of NavigationView:
```kotlin
private void initializeNavigationViewModel() {
    try {
      navigationViewModel = ViewModelProviders.of((FragmentActivity)getContext()).get(NavigationViewModel.class);
    } catch (ClassCastException exception) {
        throw new ClassCastException("Please ensure that the provided Context is a valid FragmentActivity");
    }
}
```

It is impossible to use NavigationView inside of fragment that was created with ContextThemeWrapper: 
```kotlin
override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?, savedInstanceState: Bundle?): View? {
    val wrapper = ContextThemeWrapper(context, customThemeId)
    val layoutInflater = inflater.cloneInContext(wrapper)
    return layoutInflater.inflate(layoutId, container, false)
}
```
Please, add additional check of `getBaseContext()` to make possible NavigationView usage within Fragments that inflating views with ContextThemeWrapper 

**Mapbox Navigation SDK version: 0.42.6**

**Comment by Guardiola31337** [comment]
@abhishek1508 could we check if this would be an issue in `1.0`?

**Comment by abhishek1508** [comment]
Let's fix this.
cc @JunDai @cafesilencio @asinghal22

**Comment by cafesilencio** [comment]
We were able to reproduce the issue and have fixed it in v1.0. Please be on the lookout for our first public preview of UI SDK v1.0 coming soon. Feel free to create a new issue if you feel it still exists.
